### PR TITLE
Use upstream autotools instead gnome-common ones

### PR DESCRIPTION
**Pull request by jjardon**
Follow recommendations from https://wiki.gnome.org/Projects/GnomeCommon/Migration

### CODE REVIEW

File: src/drv-iio-buffer-accel.c [hunk 1]
```diff
@@ -54,9 +54,9 @@ process_scan (IIOSensorData data, DrvData *or_data)
 		return 0;
 	}
 
-	process_scan_1(data.data + or_data->buffer_data->scan_size*i, or_data->buffer_data, "in_accel_x", &accel_x, &scale, &present_x);
-	process_scan_1(data.data + or_data->buffer_data->scan_size*i, or_data->buffer_data, "in_accel_y", &accel_y, &scale, &present_y);
-	process_scan_1(data.data + or_data->buffer_data->scan_size*i, or_data->buffer_data, "in_accel_z", &accel_z, &scale, &present_z);
+	process_scan_1(data.data + or_data->buffer_data->scan_size*i, or_data->buffer_data, (char *) "in_accel_x", &accel_x, &scale, &present_x);
```
Comment (hadess) [inline src/drv-iio-buffer-accel.c]: The correct fix might be to make process_scan_1 take a const char * instead.

File: src/iio-sensor-proxy.c [hunk 2]
```diff
@@ -734,8 +734,6 @@ sensor_changes (GUdevClient *client,
 		if (!any_sensors_left (data))
 			g_main_loop_quit (data->loop);
 	} else if (g_strcmp0 (action, "add") == 0) {
-		guint i;
```
Comment (hadess) [inline src/iio-sensor-proxy.c]: Separate patch for that, and a better description of the other warning would be appreciated.

**Comment by hadess** [comment]
Rest looks like to me.

File: src/iio-sensor-proxy.c (hunk 2 above)
Comment (jjardon) [inline src/iio-sensor-proxy.c]: Fixed

**Comment by hadess** [comment]
The warnings fix against iio-buffer-utils.h is missing a prefix to the commit subject. The correct prefix for the last commit is "main:" not "src/iio-sensor-proxy.c".

File: src/iio-buffer-utils.c [hunk 3]
```diff
@@ -248,7 +248,7 @@ build_channel_array (const char        *device_dir,
 			current->name = g_strndup (name, strlen(name) - strlen("_en"));
 			current->generic_name = iioutils_break_up_name (current->name);
 			if (g_strcmp0(current->generic_name, "in_rot_from_north_magnetic_tilt") == 0) {
-				current->generic_name = "in_rot";
+				current->generic_name = (char *) "in_rot";
```
Comment (hadess) [inline src/iio-buffer-utils.c]: This is completely wrong. The string should have been duped instead of simply cast.

**Comment by hadess** [comment]
All pushed with some commit log or style changes. except the "warnings" patch which I re-did completely. Thanks.

**Pull request closed without merging**